datafusion/functions/src/math/monotonicity.rs

```diff
@@ -558,3 +558,441 @@ pub fn get_tanh_doc() -> &'static Documentation {
         .build()
     })
 }
+
+#[cfg(test)]
+mod tests {
+    use arrow::compute::SortOptions;
+    use datafusion_common::Result;
+
+    use super::*;
+
+    #[derive(Debug)]
+    struct MonotonicityTestCase {
+        name: &'static str,
+        func: fn(&[ExprProperties]) -> Result<SortProperties>,
+        lower: f64,
+        upper: f64,
+        input_sort: SortProperties,
+        expected: Option<SortProperties>,
+        expect_err: bool,
```

Comment:
It would be nice to have the expected error be a string rather than a bool

Something like 
```suggestion
        expect_err: Option<&'static str>,
```

That way we can ensure the errors aren't changing unexpectedly

Reply:
It was like that before but due to some issues in CI and local difference. I took a shortcut there but probably should not have done that. Let me properly fix it. Thanks for the review

Reply:
Oh, I got it now. I needed to `strip_backtrace()` before comparing errors now I think I've fixed it